tests/test_devices.py

```diff
@@ -60,6 +59,135 @@ def test_device_constructor_with_config(client):
+        assert dict(alarm_device.items(q.SECTORS, status=False)) == {
+            2: {"element": 3, "excluded": False, "id": 3, "index": 2, "name": "S3 Outdoor", "status": False}
+        }
+
+    def test_without_inventory(self, alarm_device):
+        """Verify that querying items without inventory populated works correctly"""
+        alarm_device._inventory = {}
+        # Test
+        assert dict(alarm_device.items(q.SECTORS, status=False)) == {}
+
+    def test_with_empty_query(self, alarm_device):
+        """Verify that querying items with empty query works correctly"""
+        alarm_device._inventory = {q.SECTORS: {}}
+        # Test
+        assert dict(alarm_device.items(q.SECTORS, status=False)) == {}
+
+
+class TestItemAlerts:
+    def test_without_status(self, alarm_device):
+        """Verify that querying items without specifying a status works correctly"""
+        alarm_device.connect("username", "password")
+        alerts = {
+            0: {"name": "alarm_led", "status": 0},
+            1: {"name": "anomalies_led", "status": 1},
+            2: {"name": "device_failure", "status": 0},
+            3: {"name": "device_low_battery", "status": 0},
+            4: {"name": "device_no_power", "status": 0},
+            5: {"name": "device_no_supervision", "status": 0},
+            6: {"name": "device_system_block", "status": 0},
+            7: {"name": "device_tamper", "status": 1},
+            8: {"name": "gsm_anomaly", "status": 0},
+            9: {"name": "gsm_low_balance", "status": 0},
+            10: {"name": "has_anomaly", "status": False},
+            11: {"name": "input_alarm", "status": 0},
+            12: {"name": "input_bypass", "status": 0},
+            13: {"name": "input_failure", "status": 0},
+            14: {"name": "input_low_battery", "status": 0},
+            15: {"name": "input_no_supervision", "status": 0},
+            16: {"name": "inputs_led", "status": 2},
+            17: {"name": "module_registration", "status": 0},
+            18: {"name": "panel_low_battery", "status": 0},
+            19: {"name": "panel_no_power", "status": 0},
+            20: {"name": "panel_tamper", "status": 0},
+            21: {"name": "pstn_anomaly", "status": 0},
+            22: {"name": "rf_interference", "status": 0},
+            23: {"name": "system_test", "status": 0},
+            24: {"name": "tamper_led", "status": 0},
+        }
+        # Test
+        alarm_device.update()
+        assert dict(alarm_device.items(q.ALERTS)) == alerts
+
+    def test_with_status(self, alarm_device):
+        """Verify that querying items with specifying a status works correctly"""
+        alarm_device.connect("username", "password")
+        alerts_1 = {
+            1: {"name": "anomalies_led", "status": 1},
+            7: {"name": "device_tamper", "status": 1},
+        }
+        # Test
+        alarm_device.update()
+        assert dict(alarm_device.items(q.ALERTS, status=1)) == alerts_1
```

Comment:
Duplicate this test to verify what happens if you pass `True`, as we should support also the boolean statement. Unfortunately, the backend returns a data structure that is not uniform: you have 0, 1, 2, False, True as possibile values, so we should add a test for each category (meaning `int` and `bool`).